src/inference/include/openvino/runtime/properties.hpp

```diff
@@ -0,0 +1,638 @@
+ * @endcode
+ *
+ * The following code enables caching of compiled network blobs for devices where import/export is supported
+ *
+ * @code
+ * ie.set_property(ov::cache_dir("cache/")); // enables models cache
+ * @endcode
+ */
+static constexpr Key<std::string> cache_dir{"CACHE_DIR"};
+
+namespace device {
+
+/**
+ * @brief the key for setting of required device to execute on
+ * values: device id starts from "0" - first device, "1" - second device, etc
+ */
+static constexpr Key<std::string> id{"DEVICE_ID"};
+
+/**
+ * @brief Type for device Priorities config option, with comma-separated devices listed in the desired priority
+ */
+struct Priorities : public Key<std::string> {
+private:
+    template <typename H, typename... T>
+    static inline std::string concat(const H& head, T&&... tail) {
+        return head + std::string{','} + concat(std::forward<T>(tail)...);
+    }
+
+    template <typename H>
+    static inline std::string concat(const H& head) {
+        return head;
+    }
+
+public:
+    using Key<std::string>::Key;
+
+    /**
+     * @brief Constructs device priorities
+     * @tparam Args property constructor arguments types
+     * @param args property constructor arguments
+     * @return Pair of string key representation and type erased paramter.
+     */
+    template <typename... Args>
+    inline std::pair<std::string, Any> operator()(Args&&... args) const {
+        return {str(), Any{concat(std::forward<Args>(args)...)}};
+    }
+};
+
+/**
+ * @brief Device Priorities config option, with comma-separated devices listed in the desired priority
+ */
+static constexpr Priorities priorities{"MULTI_DEVICE_PRIORITIES"};
+
+/**
+ * @brief Type for key to pass set of property values to specified device
+ */
+struct Property {
+    /**
+     * @brief Constructs property
+     * @param device_name device plugin alias
+     * @param config set of property values with names
+     * @return Pair of string key representation and type erased paramter.
+     */
+    inline std::pair<std::string, Any> operator()(const std::string& device_name, const AnyMap& config) const {
+        return {device_name, config};
+    }
+
+    /**
+     * @brief Constructs property
+     * @tparam Properties Should be the pack of `std::pair<std::string, ov::Any>` types
+     * @param device_name device plugin alias
+     * @param configs Optional pack of pairs: (config parameter name, config parameter value)
+     * @return Pair of string key representation and type erased paramter.
+     */
+    template <typename... Properties>
+    inline util::EnableIfAllProperties<std::pair<std::string, Any>, Properties...> operator()(
+        const std::string& device_name,
+        Properties&&... configs) const {
+        return {device_name, AnyMap{std::pair<std::string, Any>{configs...}}};
+    }
+};
+
+/**
+ * @brief Key to pass set of property values to specified device
+ */
+static constexpr Property properties;
```

Comment:
could you please add examples in docs how it can be used?
E.g.
```c++
core.set_property(ov::device::properties("CPU", ov::enable_profiling(true)), ov::device::properties("GPU", ov::enable_profiling(false)));
```